tools/tpm2_import.c

```diff
@@ -288,9 +297,38 @@ static bool calc_sensitive_unique_data(void) {
     (X).publicArea.parameters.symDetail.sym.mode.sym = TPM2_ALG_CFB;\
     (X).publicArea.unique.sym.size = TPM2_SHA256_DIGEST_SIZE;\
 
+#define IMPORT_KEY_RSA2K_PUBLIC_AREA(X) \
+    (X).publicArea.type = TPM2_ALG_RSA; \
+    (X).publicArea.nameAlg = TPM2_ALG_SHA256;\
+    (X).publicArea.objectAttributes &= ~TPMA_OBJECT_RESTRICTED;\
+    (X).publicArea.objectAttributes |= TPMA_OBJECT_USERWITHAUTH;\
+    (X).publicArea.objectAttributes |= TPMA_OBJECT_DECRYPT;\
+    (X).publicArea.objectAttributes |= TPMA_OBJECT_SIGN_ENCRYPT;\
+    (X).publicArea.objectAttributes &= ~TPMA_OBJECT_FIXEDTPM;\
+    (X).publicArea.objectAttributes &= ~TPMA_OBJECT_FIXEDPARENT;\
+    (X).publicArea.objectAttributes &= ~TPMA_OBJECT_SENSITIVEDATAORIGIN;\
+    (X).publicArea.authPolicy.size = 0;\
+    (X).publicArea.parameters.rsaDetail.symmetric.algorithm = TPM2_ALG_NULL;\
+    (X).publicArea.parameters.rsaDetail.scheme.scheme = TPM2_ALG_NULL;\
+    (X).publicArea.parameters.rsaDetail.keyBits = RSA_2K_MODULUS_SIZE_IN_BITS;\
+    (X).publicArea.parameters.rsaDetail.exponent = 0x0;\
+    (X).publicArea.unique.rsa.size = RSA_2K_MODULUS_SIZE_IN_BYTES;\
```

Comment:
These should probably be static inlines, but don't worry about that for now, since this patch follows the convention laid out.